### PR TITLE
[Home Page Picker] Add Skip button and Close button

### PR DESCRIPTION
**Pull request by chipsnyder**
**Fixes** https://github.com/wordpress-mobile/gutenberg-mobile/issues/2717 (Skip button)

## Description
Because we now have some changes in what Navigation bar items are available per subclass, I needed to do a bit of a refactor to make this control more extensible. Instead of simulating a navigation bar and just showing/hiding the existing one, I now have to style the bar, and on iOS 12 and below, reset the adjusted styles. 

This PR does that refactoring and adds the ✖️ and `Skip` buttons on the site creation page, which was added as part of i2 discussions of the design. 

📓 The functionality for `Skip` will be added as part of https://github.com/wordpress-mobile/gutenberg-mobile/issues/2721.

## To test:
<details>
<summary><strong>To disable or enable the development version of Home Page Picker</strong></summary>

- Open the app from the build that allows FeatureFlags such as a PR build or a local development build
    - By default, the modal layout picker will be disabled.
    - From the site page:
        - Click on your Gravatar.
        - Click on App Settings.
        - Click on Debug.
        - Toggle "Site Creation: Home Page Picker" to enable or disable the picker
        - <kdb><a href="https://user-images.githubusercontent.com/3384451/97343011-5f032980-185d-11eb-9030-06c17f732fec.gif"></a></kdb>

</details>

When enabled, the first step of the Site Creation flow will be replaced with the Home Page Picker flow.

### Display
1. Navigate to the Create WordPress.com site
    - My Sites > ➕ > Create WordPress.com site
1.  **Expect** to see an ✖️ and `Skip` in the navigation bar
1. Tap the ✖️ 
1.  **Expect** The screen to dismiss
    - I'll add the functionality for the `Skip` button as part of https://github.com/wordpress-mobile/gutenberg-mobile/issues/2721 since that card is mostly to handle that functionality anyway.  

### Navigation bar style is restored
1. Navigate to the Modal Layout Picker
1. Select a layout
1. Tap Preview
- **Expect** the style of the Navigation bar to be restored from the changes on the first screen.

## Screenshots:

Expanded | Collapsed
--- | ---
<kdb><a href="https://user-images.githubusercontent.com/3384451/98141720-ede2f800-1e94-11eb-92ab-1a1cd8eba44b.png"></a></kdb> | <kdb><a href="https://user-images.githubusercontent.com/3384451/98141950-36021a80-1e95-11eb-9bf1-f69b7fd9cfa0.png"></a></kdb>

## Additional Context: 

This is a continuation of the work done in the [Home Page Picker Project](https://github.com/wordpress-mobile/gutenberg-mobile/labels/Project%3A%20Home%20Page%20Picker). 

## PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Gutenberg/Collapsable Header/CollapsableHeaderViewController.swift [hunk 1]
```diff
@@ -218,10 +231,54 @@ class CollapsableHeaderViewController: UIViewController, NoResultsViewHost {
         /* This should be overriden in a child class in order to enable support. */
     }
 
+    // MARK: - Format Nav Bar
+    private var originalNavBarAppearance: (barStyle: UIBarStyle, barTintColor: UIColor?, isTranslucent: Bool, shadowImage: UIImage?)?
+    private func cacheNavBarApperance() {
```
Comment (mkevins) [inline WordPress/Classes/ViewRelated/Gutenberg/Collapsable Header/CollapsableHeaderViewController.swift]: NP: typo

File: WordPress/Classes/ViewRelated/Gutenberg/Collapsable Header/CollapsableHeaderViewController.swift [hunk 2]
```diff
@@ -218,10 +231,54 @@ class CollapsableHeaderViewController: UIViewController, NoResultsViewHost {
...
+        if #available(iOS 13.0, *) {
+            let newAppearance = UINavigationBarAppearance()
+            newAppearance.configureWithTransparentBackground()
+            newAppearance.backgroundColor = .clear
+            newAppearance.shadowColor = .clear
+            newAppearance.shadowImage = UIImage()
+            navigationItem.standardAppearance = newAppearance
+            navigationItem.scrollEdgeAppearance = newAppearance
+            navigationItem.compactAppearance = newAppearance
+            setNeedsStatusBarAppearanceUpdate()
```
Comment (mkevins) [inline WordPress/Classes/ViewRelated/Gutenberg/Collapsable Header/CollapsableHeaderViewController.swift]: Can this be after the conditional?

Also, I was thinking maybe the version conditional can benefit from a comment. Wdyt?

**Review by mkevins — COMMENTED** [review]
Looks good Chip, and works great! Just had one suggestion, and a typo nitpick :smile: .

Tested via the steps described, and it works as expected.

File: WordPress/Classes/ViewRelated/Gutenberg/Collapsable Header/CollapsableHeaderViewController.swift (hunk 2 above)
Comment (chipsnyder) [reply WordPress/Classes/ViewRelated/Gutenberg/Collapsable Header/CollapsableHeaderViewController.swift]: That's a good suggestion. I added some inline documentation and included `@available(iOS, obsoleted: 13.0, message: "See description on 'formatNavigationController'")` This won't actually prevent you from calling the function in iOS 13+ yet but provides some more documentation on the intended use. How does that look to you?

Comment (mkevins) [reply WordPress/Classes/ViewRelated/Gutenberg/Collapsable Header/CollapsableHeaderViewController.swift]: Awesome :+1:

**Review by mkevins — APPROVED** [review]
Great work Chip, LGTM!

**Review by antonis — APPROVED** [review]
Great work Chip 👍 
Tested on an iPhone and iPad simulator. LGTM 🎉

**Pull request merged**